karibu-dsl-v23/src/main/kotlin/com/github/mvysny/karibudsl/v23/Dialogs.kt

```diff
@@ -82,6 +85,22 @@ public fun ConfirmDialog.setCloseOnCancel(buttonText: String = karibuDslI18n("ca
     setCancelButton(buttonText) { close() }
 }
 
+/**
+ * Adds a cancel clickable component (e.g. Button) to the dialog; the dialog is closed when the button is clicked.
+ * @param clickableComponent custom clickable component.
+ *
+ * Example of usage:
+ * ```kotlin
+ * setCloseOnCancel(buildSingleComponent { button("Cancel", VaadinIcon.ARROW_BACKWARD.create()) })
```

Comment:
This example is very complicated and I'd say it's a misuse of DSL. Far simpler code is produced by simply calling the `Button()` constructor.

Reply:
See https://github.com/mvysny/karibu-tools/pull/11